tests/src/core/testqgslayoutmanualtable.cpp

```diff
@@ -530,5 +533,137 @@ void TestQgsLayoutManualTable::cellTextAlignment()
   QGSVERIFYLAYOUTCHECK( QStringLiteral( "manualtable_textalign" ), &l );
 }
 
+void TestQgsLayoutManualTable::mergedCells()
```

Comment:
Could we have a couple more checks here to test the table background logics? Thinking mostly of:
- if we have a last column color set to green, and the last column turns out to be a merged cell, will it be colored accordingly? 
- if we have an odd/even color for rows, will the merged cell adopt the first merged cell's color?

Reply:
It's *always* taken from the first (top-left) cell in the case of merged cells. I think that logic makes sense -- what do you think?

Reply:
@nyalldawson , I'm 90% sure it's a fine logic, but then there's the last column/row situation where I feel it might be better to take the last (bottom-right) cell / position.

Reply:
@nirvn 
Ok, done!

Reply:
@nyalldawson , thanks -- the additional test clearly shows that logic tweak was the right thing to do! :wink: